backend/authentication/serializers.py

```diff
@@ -1,23 +1,89 @@
-from rest_framework import serializers
-
-from .models import *
-
-
-class UserSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = User
-        fields = (
-            "id",
-            "username",
-            "name",
-            "password",
-            "location",
-            "description",
-            "verified",
-            "verification_method",
-            "verification_partner",
-            "social_accounts",
-            "total_flags",
-            "creation_date",
-            "deletion_date",
-        )
+from rest_framework import serializers
+from .models import *
+from content.models import Resource, Task, Topic
+from utils.utils import validate_creation_and_deletion_dates, validate_empty, validate_object_existence
+class SupportEntityTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SupportEntityType
+        fields = '__all__'
+
+    def validate(self, data):
+        validate_empty(data["name"], "name")
+
+        if len(data["name"]) < 3:
+            raise serializers.ValidationError(
+                "name must be at least 3 characters long"
+            )
+        return data
+class SupportSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Support
+        fields = '__all__'
+
+    def validate(self, data):
+        
+        if data["supporter_type"] == data["supported_type"]:

```

Comment:
This one isn't needed as they can be the same if say a user is supporting another user. The entity cannot be the same for sure, but then for this we need to keep in mind that we need to not repeat ids for organizations for those of users :)

I can remove this. No stress! 😎